src/quic/node_quic_socket.cc

```diff
@@ -457,25 +484,30 @@ void QuicSocket::OnReceive(
   QuicCID dcid(pdcid, pdcidlen);
   QuicCID scid(pscid, pscidlen);
 
+  // TODO(@jasnell): It would be fantastic if Debug() could be
+  // modified to accept objects with a ToString-like capability
+  // similar to what we can do with TraceEvents... that would
+  // allow us to pass the QuicCID directly to Debug and have it
+  // converted to hex only if the category is enabled so we can
+  // skip committing resources here.
```

Comment:
Fully agree :+1: 

It’s a bit tricky because that means we can’t use `fprintf()` directly anymore, but that’s probably pretty okay actually given that having a C++-y replacement would also help solve issues like https://github.com/nodejs/node/issues/28761.